frontend-v2/src/features/data/LoadData.tsx

```diff
@@ -113,6 +113,10 @@ const LoadData: FC<ILoadDataProps> = ({ state, firstTime }) => {
     setMinimumInfusionTime(state);
   }
 
+  useEffect(() => {
+    setShowData(state.data.length > 0 && state.fields.length > 0)
+  }, [state.data.length, state.fields.length])
```

Comment:
`useEffect` shouldn't really be used to set component state: See https://react.dev/learn/you-might-not-need-an-effect

Is this effect here because the state change is handled by an event handler in the parent component?